.github/workflows/build_and_deploy_docs.yaml

```diff
@@ -0,0 +1,94 @@
+name: Build and deploy docs to Github Pages
+
+on:
+  push:
+    branches:
+      - dev
+
+jobs:
+  build-saturn:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Log details
+        run: |
+          BRANCH=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}
+          echo "Triggered on branch: $BRANCH"
+
+      - name: Set up Ruby (required for gem installation)
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '2.7'
+
+      - name: Install Asciidoctor to build docs
+        run: |
+          gem install asciidoctor
+          gem install asciidoctor-pdf
+          gem install rouge
+
+      - name: Build Docs - Collect needed files
+        run: |
+          set -e
+          PROJECT_FILES=(
+            "projects/saturn/src/main/resources/log4j2.properties"
+            "projects/saturn/src/main/resources/system-vocabulary.ttl"
+            "projects/saturn/taxonomies.ttl"
+            "projects/saturn/views.yaml"
+            "projects/saturn/vocabulary.ttl"
+          )
+          BUILD_DIR=./docs/build
+          version=$(cat VERSION)
+          
+          mkdir -p $BUILD_DIR/docs
+          cp ./README.adoc $BUILD_DIR
+          sed -i -e "s/VERSION/${version}/" $BUILD_DIR/README.adoc
+          cp -r ./docs/images $BUILD_DIR/docs/
+          for f in ${PROJECT_FILES[*]}; do
+          mkdir -p "$BUILD_DIR/$(dirname "$f")"
+          cp "$f" "$BUILD_DIR/"$(dirname "$f")""
+          done
+
+      - name: Build Docs - Generate PDF and HTML
+        run: |
+          set -e
+          BUILD_DIR=./docs/build
+          asciidoctor-pdf -a pdf-theme=./docs/pdf-theme.yml -o $BUILD_DIR/Fairspace.pdf $BUILD_DIR/README.adoc || {
+            echo "Error building PDF"
+            popd
+            exit 1
+          }
+          asciidoctor -a toc=left -D $BUILD_DIR/ -o index.html $BUILD_DIR/README.adoc || {
+            echo "Error building site"
+            popd
+            exit 1
+          }
+          rm $BUILD_DIR/README.adoc
+
+      - name: Deploy Docs (push to Github Pages, will be deployed automatically)
+        env:
+          CI_SERVICE_ACCOUNT_USER: ${{ secrets.CI_SERVICE_ACCOUNT_USER }}
+          CI_SERVICE_ACCOUNT_PASSWORD: ${{ secrets.FNS_PAT }}
+          DOCUMENTATION_REPO: ${{ vars.DOCUMENTATION_REPO }}
+        run: |
+          set -e
+          DOCS_REPO="https://${CI_SERVICE_ACCOUNT_USER}:${CI_SERVICE_ACCOUNT_PASSWORD}@github.com/thehyve/${DOCUMENTATION_REPO}"
```

Comment:
`DOCS_REPO` and `DOCUMENTATION_REPO` are not clear variable names now. Need to scroll to this line to know which is which each time. Could you name one `DOCUMENTATION_REPO_NAME` or the other more like `..._FULL_PATH` or something more descriptive?

Reply:
makes sense, renamed them consistently